SmartDeviceLink/SDLFileManager.m

```diff
@@ -209,16 +210,126 @@ - (void)deleteRemoteFileWithName:(SDLFileName *)name completionHandler:(nullable
                                                                                  }
 
                                                                                  if (handler != nil) {
-                                                                                     handler(YES, self.bytesAvailable, nil);
+                                                                                     handler(success, self.bytesAvailable, error);
                                                                                  }
                                                                              }];
 
     [self.transactionQueue addOperation:deleteOperation];
 }
 
+- (void)deleteRemoteFilesWithNames:(NSArray<SDLFileName *> *)names completionHandler:(nullable SDLFileManagerMultiDeleteCompletionHandler)completionHandler {
+    if (names.count == 0) {
+        return completionHandler([NSError sdl_fileManager_noFilesError]);
```

Comment:
We should throw an exception here instead. This is a programmer error, not one that occurs "accidentally" in production.